app/jobs/maintenance_tasks/task_job.rb

```diff
@@ -97,8 +97,23 @@ def on_shutdown
       @ticker.persist
     end
 
+    # We are reopening a private part of Job Iteration's API here, so we should
+    # ensure the method is still defined upstream. This way, in the case where
+    # the method changes upstream, we catch it at load time instead of at
+    # runtime while calling `super`.
+    unless private_method_defined?(:reenqueue_iteration_job)
+      raise 'JobIteration::Iteration#reenqueue_iteration_job must be defined'
+    end
+    def reenqueue_iteration_job(should_ignore: true)
```

Comment:
Technically we're not ignoring it, more like postponing it but ok 😅

Reply:
I thought about using "postpone" instead, but then figured it's not really postponed because we call it again explicitly, it is more just outright ignored the first time 😛